src/Microsoft.Health.Fhir.R4.Core/Features/Operations/Export/ExportAnonymizerFactory.cs

```diff
@@ -0,0 +1,63 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Anonymizer.Core;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Export
+{
+    public class ExportAnonymizerFactory : IAnonymizerFactory
+    {
+        private IArtifactProvider _artifactProvider;
+        private ILogger<ExportJobTask> _logger;
+
+        public ExportAnonymizerFactory(IArtifactProvider artifactProvider, ILogger<ExportJobTask> logger)
+        {
+            _artifactProvider = artifactProvider;
+            _logger = logger;
+        }
+
+        public async Task<IAnonymizer> CreateAnonymizerAsync(string configurationLocation, CancellationToken cancellationToken)
+        {
+            using (Stream stream = new MemoryStream())
+            {
+                try
+                {
+                    await _artifactProvider.FetchAsync(configurationLocation, stream, cancellationToken);
+                    stream.Position = 0;
+                }
+                catch (FileNotFoundException ex)
+                {
+                    _logger.LogError($"Anonymization configuration file not found: {configurationLocation}");
+                    throw new AnonymizationConfigurationNotFoundException(ex.Message, ex);
```

Comment:
It seems that `$"Anonymization configuration file not found: {configurationLocation}"` should be the exception message. 
I think we've log the valuable message then throw an AnonymizationConfigurationNotFoundException with the original exception message which probably isn't as useful. Also given you're passing in the inner exception the original message will already be in there

Reply:
Yes, I already put these information in inner exception. Updated.